apps/examples/st_things/sample_light_user.c

```diff
@@ -0,0 +1,93 @@
+/* ****************************************************************
+*
+* Copyright 2017 Samsung Electronics All Rights Reserved.
+*
+*
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+******************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <inttypes.h>
+#include <signal.h>
+#include <st_things/st_things.h>
+#include <tinyara/gpio.h>
+#include <fcntl.h>
+
+#define TAG                     "LIGHT_USER"
+#define BLUE_LED 49
```

Comment:
same as previous

Reply:
@KimDongEon

Reply:
This is an example of ARTIK053. and
This example is dependent on ARTIK053.

Reply:
So, why do you make a dependancy with ARTIK053? When we add new board, then it will not work. But st_things doesn't have any board dependancy. Make a new configs to get those values. This is a common example.

Reply:
I understand what you mean. He'll use condition statement, so 'button' only use when board is artik053. Then I'll change the Kconfig to remove artik053 dependenccy. Thanks.

Reply:
Don't do like belows.
```
#if artik053
#define button 49
#else
#define button
#endif
```
Adding board specifc codes is bad in common codes.
Here is an example.
```
config <EXAMPLE_NAME>_SUPPORT_BUTTON

if <EXAMPLE_NAME>_SUPPORT_BUTTON
config <EXAMPLE_NAME>_BUTTON_GPIO_NUM
endif

#if config <EXAMPLE_NAME>_SUPPORT_BUTTON
#define button config <EXAMPLE_NAME>_SUPPORT_BUTTON_GPIO_NUM
#else
#define button 
#endif

They should be defined in artik053 defconfig, 
```

Reply:
I will remove